src/celeritas/global/CoreTrackData.hh

```diff
@@ -113,6 +116,7 @@ struct CoreStateData
     SimStateData<W, M> sim;
     TrackInitStateData<W, M> init;
     ThreadItems<TrackSlotId::size_type> track_slots;
+    ActionItems<typename decltype(track_slots)::ItemIdT> thread_offsets;
```

Comment:
This is invalidated if `track_slots` is no longer sorted for any reason (added new tracks, updated action,...)